### ISSUE TITLE
Stage3 - Error Synchronizing

### ISSUE THREAD

**Issue by lcordero** [konveyor/pvc-migrate#136]
**Description:**
Rsync is using localhost instead of using the destination route for rsync.

**Error:**
```
TASK [Synchronizing files. This may take a while...] *****************************************************
fatal: [localhost -> d-worker04.in.appcanvas.net]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": true, "cmd": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ab7dc8d1-ea2f-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-d76cbd60-e966-11ea-a05d-0273698e724c/ rsync://root@localhost/lcordero-test-huge/", "delta": "0:00:00.022956", "end": "2020-09-03 03:45:29.878927", "msg": "non-zero return code", "rc": 5, "start": "2020-09-03 03:45:29.855971", "stderr": "@ERROR: access denied to lcordero-test-huge from localhost (127.0.0.1)\nrsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]", "stderr_lines": ["@ERROR: access denied to lcordero-test-huge from localhost (127.0.0.1)", "rsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]"], "stdout": "opening tcp connection to localhost port 2222\nsending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . lcordero-test-huge/  (6 args)", "stdout_lines": ["opening tcp connection to localhost port 2222", "sending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . lcordero-test-huge/  (6 args)"]}
```

**Comment by jwmatthews** [comment]
@pranavgaikwad please take a look and comment

**Comment by pranavgaikwad** [comment]
@lcordero `localhost` is expected since stunnel is doing the job of forwarding the connection to the destination route. Stunnel accepts connection on localhost 2222 port. See this configuration here : [https://github.com/konveyor/pvc-migrate/blob/3f4bd6a4ef77792e1fce08fd51cf77f17844e527/3_run_rsync/tasks/rsync.yml#L45-L58](https://github.com/konveyor/pvc-migrate/blob/3f4bd6a4ef77792e1fce08fd51cf77f17844e527/3_run_rsync/tasks/rsync.yml#L45-L58)

Now, one thing I find strange in your output is that the `rsync` task might not be running as root. It should actually run as privileged as done here : [https://github.com/konveyor/pvc-migrate/blob/3f4bd6a4ef77792e1fce08fd51cf77f17844e527/3_run_rsync/tasks/rsync.yml#L137](https://github.com/konveyor/pvc-migrate/blob/3f4bd6a4ef77792e1fce08fd51cf77f17844e527/3_run_rsync/tasks/rsync.yml#L137)

Can you please confirm if it is actually running as root? You can find out by setting `-vvv` verbosity options on `ansible-playbook` command and look for `Escalation succeded` message in  the output of rsync task:

```
<node2.pgaikwad-o3.internal> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/pranav/.ansible/cp/7bb46bb9a1 -tt node2.pgaikwad-o3.internal '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-mmbkzjhqpzvkferbhdgmdzuuxhasfhyc ; RSYNC_PASSWORD=RyRwqDYthGNqeoH /usr/bin/python /home/ec2-user/.ansible/tmp/ansible-tmp-1599151502.837387-2040694-26992045224497/AnsiballZ_command.py'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
```

**Comment by roleymoley** [comment]
Hi @pranavgaikwad Luckily we ran with vvv 

```
TASK [Synchronizing files. This may take a while...] ******************************************
task path: /home/ec2-user/PVC-migrate-1-1-2/pvc-migrate/3_run_rsync/tasks/rsync.yml:132
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthen
tications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08
.in.appcanvas.net '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'/home/ec2-user\n', b'')
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthen
tications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08
.in.appcanvas.net '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/ec2-user/.ansible/tmp `"&& mkdir /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-237
63704473814 && echo ansible-tmp-1599815055.085095-17521-23763704473814="` echo /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814 `" ) && sleep
0'"'"''
<d-worker08.in.appcanvas.net> (0, b'ansible-tmp-1599815055.085095-17521-23763704473814=/home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814\n', b'
')
<localhost> Attempting python interpreter discovery
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthen
tications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08
.in.appcanvas.net '/bin/sh -c '"'"'echo PLATFORM; uname; echo FOUND; command -v '"'"'"'"'"'"'"'"'/usr/bin/python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.7'"'"
'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.5'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.7'"'"'"'"
'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/usr/libexec/platform-python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/u
sr/bin/python3'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python'"'"'"'"'"'"'"'"'; echo ENDFOUND && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'PLATFORM\nLinux\nFOUND\n/usr/bin/python\n/usr/bin/python2.7\n/usr/libexec/platform-python\n/usr/bin/python\nENDFOUND\n', b'')
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthen
tications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08
.in.appcanvas.net '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'{"osrelease_content": "NAME=\\"Red Hat Enterprise Linux Server\\"\\nVERSION=\\"7.8 (Maipo)\\"\\nID=\\"rhel\\"\\nID_LIKE=\\"fedora\\"\\nV
ARIANT=\\"Server\\"\\nVARIANT_ID=\\"server\\"\\nVERSION_ID=\\"7.8\\"\\nPRETTY_NAME=\\"OpenShift Enterprise\\"\\nANSI_COLOR=\\"0;31\\"\\nCPE_NAME=\\"cpe:/o:redhat:enterprise
_linux:7.8:GA:server\\"\\nHOME_URL=\\"https://www.redhat.com/\\"\\nBUG_REPORT_URL=\\"https://bugzilla.redhat.com/\\"\\n\\nREDHAT_BUGZILLA_PRODUCT=\\"Red Hat Enterprise Linu
x 7\\"\\nREDHAT_BUGZILLA_PRODUCT_VERSION=7.8\\nREDHAT_SUPPORT_PRODUCT=\\"Red Hat Enterprise Linux\\"\\nREDHAT_SUPPORT_PRODUCT_VERSION=\\"7.8\\"\\n", "platform_dist_result":
 ["redhat", "7.8", "Maipo"]}\n', b'')
Using module file /home/ec2-user/pvc-migrate/2_pvc_destination_gen/env/lib64/python3.6/site-packages/ansible/modules/commands/command.py
<d-worker08.in.appcanvas.net> PUT /home/ec2-user/.ansible/tmp/ansible-local-14424yng621qq/tmpslv96xda TO /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-237
63704473814/AnsiballZ_command.py
<d-worker08.in.appcanvas.net> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o Preferred
Authentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.an
sible/cp/811f235127 '[d-worker08.in.appcanvas.net]'
<d-worker08.in.appcanvas.net> (0, b'sftp> put /home/ec2-user/.ansible/tmp/ansible-local-14424yng621qq/tmpslv96xda /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/AnsiballZ_command.py\n', b'')
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08.in.appcanvas.net '/bin/sh -c '"'"'chmod u+x /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/ /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/AnsiballZ_command.py && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'', b'')
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 -tt d-worker08.in.appcanvas.net '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-uuszxdyolnjtvvdwemovstmpkbzjwjas ; RSYNC_PASSWORD=twFXwqadVchEykj /usr/bin/python /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/AnsiballZ_command.py'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<d-worker08.in.appcanvas.net> (1, b'\r\n{"changed": true, "end": "2020-09-11 14:34:15.496229", "stdout": "opening tcp connection to localhost port 2222\\nsending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . postgresql/  (6 args)", "cmd": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ed44077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-131e-11e9-9098-0a15905bf50e/ rsync://root@localhost/postgresql/", "failed": true, "delta": "0:00:00.025510", "stderr": "@ERROR: access denied to postgresql from localhost (127.0.0.1)\\nrsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]", "rc": 5, "invocation": {"module_args": {"creates": null, "executable": null, "_uses_shell": true, "strip_empty_ends": true, "_raw_params": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ed44077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-131e-11e9-9098-0a15905bf50e/ rsync://root@localhost/postgresql/", "removes": null, "argv": null, "warn": true, "chdir": null, "stdin_add_newline": true, "stdin": null}}, "start": "2020-09-11 14:34:15.470719", "msg": "non-zero return code"}\r\n', b'Shared connection to d-worker08.in.appcanvas.net closed.\r\n')
<d-worker08.in.appcanvas.net> Failed to connect to the host via ssh: Shared connection to d-worker08.in.appcanvas.net closed.
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08.in.appcanvas.net '/bin/sh -c '"'"'rm -f -r /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/ > /dev/null 2>&1 && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'', b'')
fatal: [localhost -> d-worker08.in.appcanvas.net]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
    "cmd": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ed44077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-131e-11e9-9098-0a15905bf50e/ rsync://root@localhost/postgresql/",
    "delta": "0:00:00.025510",
    "end": "2020-09-11 14:34:15.496229",
    "invocation": {
        "module_args": {
            "_raw_params": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ed44077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-131e-11e9-9098-0a15905bf50e/ rsync://root@localhost/postgresql/",
            "_uses_shell": true,
            "argv": null,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "stdin_add_newline": true,
            "strip_empty_ends": true,
            "warn": true
        }
    },
    "msg": "non-zero return code",
    "rc": 5,
    "start": "2020-09-11 14:34:15.470719",
    "stderr": "@ERROR: access denied to postgresql from localhost (127.0.0.1)\nrsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]",
    "stderr_lines": [
        "@ERROR: access denied to postgresql from localhost (127.0.0.1)",
        "rsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]"
    ],
    "stdout": "opening tcp connection to localhost port 2222\nsending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . postgresql/  (6 args)",
    "stdout_lines": [
        "opening tcp connection to localhost port 2222",
        "sending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . postgresql/  (6 args)"
    ]
}
Read vars_file 'vars/run-rsync.yml'
Read vars_file 'vars/defaults.yml'
Read vars_file 'vars/run-rsync.yml'
Read vars_file 'vars/defaults.yml'
```
This is the full output we have for that task and it looks like it does escalate successfully to me

**Comment by pranavgaikwad** [comment]
@roleymoley The `become` directive should ask Ansible to use `root` user. However, in your case, it looks like its using `postgresql` user.

**Comment by pranavgaikwad** [comment]
@roleymoley do you by any chance have a ansible.cfg or environment variable set that sets `become_user` or `ansible_become_user` to `postgresql`?

**Comment by pranavgaikwad** [comment]
Reopening as RC was wrong. See https://github.com/konveyor/pvc-migrate/issues/137#issuecomment-695081277 for explanation. 

Fixed through #141
